service/src/main/java/bio/terra/workspace/app/controller/ControlledFlexibleResourceApiController.java

```diff
@@ -0,0 +1,130 @@
+      AuthenticatedUserRequestFactory authenticatedUserRequestFactory,
+      HttpServletRequest request,
+      ControlledResourceService controlledResourceService,
+      SamService samService,
+      WorkspaceService workspaceService,
+      ControlledResourceMetadataManager controlledResourceMetadataManager) {
+    super(authenticatedUserRequestFactory, request, controlledResourceService, samService);
+    this.workspaceService = workspaceService;
+    this.controlledResourceService = controlledResourceService;
+    this.controlledResourceMetadataManager = controlledResourceMetadataManager;
+  }
+
+  @Traced
+  @Override
+  public ResponseEntity<ApiCreatedControlledFlexibleResource> createFlexibleResource(
+      UUID workspaceUuid, @Valid ApiCreateControlledFlexibleResourceRequestBody body) {
+    final AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+
+    ControlledResourceFields commonFields =
+        toCommonFields(
+            workspaceUuid,
+            body.getCommon(),
+            null,
+            userRequest,
+            WsmResourceType.CONTROLLED_FLEXIBLE_RESOURCE);
+
+    byte[] encodedJSON = body.getFlexibleResource().getData();
+    String decodedJSON =
+        encodedJSON != null ? new String(encodedJSON, StandardCharsets.UTF_8) : null;
+
+    ControlledFlexibleResource resource =
+        ControlledFlexibleResource.builder()
+            .common(commonFields)
+            .typeNamespace(body.getFlexibleResource().getTypeNamespace())
+            .type(body.getFlexibleResource().getType())
+            .data(decodedJSON)
+            .build();
+
+    ControlledFlexibleResource createdFlexibleResource =
+        getControlledResourceService()
+            .createControlledResourceSync(resource, commonFields.getIamRole(), userRequest, body)
+            .castByEnum(WsmResourceType.CONTROLLED_FLEXIBLE_RESOURCE);
+
+    var response =
+        new ApiCreatedControlledFlexibleResource()
+            .resourceId(createdFlexibleResource.getResourceId())
+            .flexibleResource(createdFlexibleResource.toApiResource());
+
+    return new ResponseEntity<>(response, HttpStatus.OK);
+  }
+
+  @Traced
+  @Override
+  public ResponseEntity<Void> deleteFlexibleResource(UUID workspaceUuid, UUID resourceId) {
+    final AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+    controlledResourceMetadataManager.validateControlledResourceAndAction(
+        userRequest,
+        workspaceUuid,
+        resourceId,
+        SamConstants.SamControlledResourceActions.DELETE_ACTION);
+    logger.info(
+        "Deleting controlled flexible resource {} in workspace {}",
+        resourceId.toString(),
+        workspaceUuid.toString());
+    getControlledResourceService()
+        .deleteControlledResourceSync(workspaceUuid, resourceId, userRequest);
+    return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+  }
+
+  @Traced
+  @Override
+  public ResponseEntity<ApiFlexibleResource> getFlexibleResource(
```

Comment:
now that we have this endpoint /api/workspaces/v1/{workspaceId}/resources/{resourceId}

do we still need this? @ddietterich

Reply:
It keeps things consistent, so I'm happy to have it.